appinventor/components/src/com/google/appinventor/components/runtime/ChartDataBase.java

```diff
@@ -108,13 +111,28 @@ public YailList Colors() {
      */
     @SimpleProperty
     public void Colors(YailList argb) {
-        colors = argb;
-
-        // TODO: Add support for other Chart types
-        if (chartDataModel instanceof PieChartDataModel) {
-            ((PieChartDataModel)chartDataModel).setColors(argb);
+        // Parse the entries of the YailList
+        List<Integer> resultColors = new ArrayList<Integer>();
+
+        for (int i = 0; i < argb.size(); ++i) {
+            // Get the element of the YailList as a String
+            String color = argb.getString(i);
+
+            try {
+                // Parse the color value and add it to the results List
+                int colorValue = Integer.parseInt(color);
+                resultColors.add(colorValue);
+            } catch (NumberFormatException e) {
+                // Skip invalid entry
+            }
         }
 
+        // Update the Colors YailList variable
+        colors = YailList.makeList(resultColors);
+
+        // Set the colors from the constructed List of colors
+        // and refresh the Chart.
+        chartDataModel.setColors(resultColors);
```

Comment:
Test with 2 Charts:
1.Set colors of Chart1
2.Get colors from Chart1
3.Set colors to Chart2